resources/components/base/Button.vue

```diff
@@ -0,0 +1,246 @@
+<template>
+	<button
+		class="wbmad-button"
+		v-bind:class="classObject"
+		v-bind:disabled="disabled"
+		v-bind:title="title"
+		v-on:click="callback"
+	>
+		<icon
+			v-if="icon"
+			v-bind:icon="icon"
+			v-bind:invert="invert"
+		/>
+		<slot />
+	</button>
+</template>
+
+<script>
+var Icon = require( './Icon.vue' );
+
+module.exports = {
+	name: 'Button',
+
+	components: {
+		icon: Icon
+	},
+
+	props: {
+		callback: {
+			type: Function,
+			required: true
+		},
+		classes: {
+			type: Array,
+			default: []
+		},
+		title: {
+			type: String,
+			default: null
+		},
+		disabled: {
+			type: Boolean,
+			default: false
+		},
+		framed: {
+			type: Boolean,
+			default: true
+		},
+		icon: {
+			type: String,
+			default: null
+		},
+		// In OOUI, flags are passed in as an array (or a string or an object)
+		// and are handled by a separate mixin. Passing them in individually is
+		// a bit more readable and intuitive, plus it makes the code in this
+		// component simpler.
+		progressive: {
+			type: Boolean,
+			default: false
+		},
+		destructive: {
+			type: Boolean,
+			default: false
+		},
+		primary: {
+			type: Boolean,
+			default: false
+		}
+	},
+
+	computed: {
+		classObject() {
+			var customClasses = {};
```

Comment:
Is there a better way to include custom classnames here @egardner? We should discuss strategy for including custom classes and things like flags...

Reply:
See above comment – ideally custom classes would just live in the template where the component is used.

I think a computed property that returns a class object is still a good idea here, but per the comments above we can probably simplify it. I'd remove the custom class part entirely (since that can be provided in parent context), which means we don't need `$.extend()` here (man, I miss `...`). Instead the computed property should just return an object with the appropriate built-in classes set as either true or false.